terraform/jenkins/jenkins_controller.tf

```diff
@@ -0,0 +1,148 @@
+# SPDX-FileCopyrightText: 2023 Technology Innovation Institute (TII)
+#
+# SPDX-License-Identifier: Apache-2.0
+
+# Build the Jenkins controller image
+module "jenkins_controller_image" {
+  source = "../../tf-modules/azurerm-nix-vm-image"
+
+  nix_attrpath   = "outputs.nixosConfigurations.jenkins-controller.config.system.build.azureImage"
+  nix_entrypoint = "${path.module}/../.."
+
+
+  name                = "jenkins-controller"
+  resource_group_name = azurerm_resource_group.default.name
+  location            = azurerm_resource_group.default.location
+
+  storage_account_name   = azurerm_storage_account.vm_images.name
+  storage_container_name = azurerm_storage_container.vm_images.name
+}
+
+# Create a machine using this image
+module "jenkins_controller_vm" {
+  source = "../../tf-modules/azurerm-linux-vm"
+
+  resource_group_name = azurerm_resource_group.default.name
+  location            = azurerm_resource_group.default.location
+
+  virtual_machine_name         = "ghaf-jenkins-controller"
+  virtual_machine_size         = "Standard_D1_v2"
+  virtual_machine_source_image = module.jenkins_controller_image.image_id
+
+  virtual_machine_custom_data = join("\n", ["#cloud-config", yamlencode({
+    users = [
+      for user in toset(["bmg", "flokli", "hrosten"]) : {
+        name                = user
+        sudo                = "ALL=(ALL) NOPASSWD:ALL"
+        ssh_authorized_keys = local.ssh_keys[user]
+      }
+    ]
+    write_files = [
+      # See corresponding EnvironmentFile= directives in services
+      {
+        content = "KEY_VAULT_NAME=${azurerm_key_vault.ssh_remote_build.name}\nSECRET_NAME=${azurerm_key_vault_secret.ssh_remote_build.name}",
+        "path"  = "/var/lib/fetch-build-ssh-key/env"
+      },
+      {
+        content = "KEY_VAULT_NAME=${azurerm_key_vault.binary_cache_signing_key.name}\nSECRET_NAME=${azurerm_key_vault_secret.binary_cache_signing_key.name}",
+        "path"  = "/var/lib/fetch-binary-cache-signing-key/env"
```

Comment:
and again the signing key traveling somewhere ...

Reply:
It's not travelling, this is configuring the `fetch-binary-cache-signing-key` service, so it knows where to fetch the key from to put on disk.

Regarding "putting on disk", I refer to https://github.com/tiiuae/ghaf-infra/pull/35#discussion_r1432769524.

Reply:
needs further consideration after holidays.